tests/functional/olp-cpp-sdk-dataservice-read/ApiTest.cpp

```diff
@@ -89,16 +89,10 @@ class ApiTest : public ::testing::Test {
 TEST_F(ApiTest, GetCatalog) {
   olp::client::HRN hrn(GetTestCatalog());
 
-  std::promise<olp::dataservice::read::ApiClientLookup::ApiClientResponse>
-      config_client_promise;
-  olp::dataservice::read::ApiClientLookup::LookupApiClient(
-      olp::client::OlpClientFactory::Create(*settings_), "config", "v1", hrn,
-      [&config_client_promise](
-          olp::dataservice::read::ApiClientLookup::ApiClientResponse response) {
-        config_client_promise.set_value(response);
-      });
-
-  auto client_response = config_client_promise.get_future().get();
+  auto client_response = olp::dataservice::read::ApiClientLookup::LookupApi(
+      hrn, {}, "config", "v1",
+      olp::dataservice::read::FetchOptions::OnlineIfNotFound, *settings_);
```

Comment:
```suggestion
      olp::dataservice::read::FetchOptions::OnlineOnly, *settings_);
```
As I understand, the previous version of this test that you deleted was always using Online mode, wasn't it? 
Anyway, it is better to cover all fetch options in the tests (integration / unit) as a separate ticket.